src/awx_plugins/credentials/aws_assumerole.py

```diff
@@ -0,0 +1,128 @@
+import datetime
+import hashlib
+
+from django.utils.translation import gettext_lazy as _
+
+import boto3
+
+from .plugin import CredentialPlugin
+
+
+try:
+    from botocore.exceptions import ClientError
+except ImportError:
+    """Caught by AnsibleAWSModule."""
+
+_aws_cred_cache = {}
+
+
+assume_role_inputs = {'fields': [{'id': 'access_key',
+                                  'label': _('AWS Access Key'),
+                                  'type': 'string',
+                                  'secret': True,
+                                  'help_text': _('The optional AWS access key for the user who will assume the role'),
+                                  },
+                                 {'id': 'secret_key',
+                                  'label': 'AWS Secret Key',
+                                  'type': 'string',
+                                  'secret': True,
+                                  'help_text': _('The optional AWS secret key for the user who will assume the role'),
+                                  },
+                                 {'id': 'external_id',
+                                  'label': 'External ID',
+                                  'type': 'string',
+                                  'help_text': _('The optional External ID which will be provided to the assume role API'),
+                                  },
+                                 {'id': 'role_arn',
+                                  'label': 'AWS ARN Role Name',
+                                  'type': 'string',
+                                  'secret': True,
+                                  'help_text': _('The ARN Role Name to be assumed in AWS'),
+                                  },
+                                 ],
+                      'metadata': [{'id': 'identifier',
+                                    'label': 'Identifier',
+                                    'type': 'string',
+                                    'help_text': _('The name of the key in the assumed AWS role to fetch [AccessKeyId | SecretAccessKey | SessionToken].'),
+                                    },
+                                   ],
+                      'required': ['role_arn'],
+                      }
+
+
+def aws_assumerole_getcreds(access_key, secret_key, role_arn, external_id):
+    if (access_key is None or len(access_key) == 0) and (
+            secret_key is None or len(secret_key) == 0
+    ):
+        # Connect using credentials in the EE
+        connection = boto3.client(service_name='sts')
+    else:
+        # Connect to AWS using provided credentials
+        connection = boto3.client(
+            service_name='sts',
+            aws_access_key_id=access_key,
+            aws_secret_access_key=secret_key,
+        )
```

Comment:
Would this be simpler?
```suggestion
    explicit_credentials_empty = not access_key and not secret_key
    credential_kwargs = {} if explicit_credentials_empty else {
        # EE creds are read from the env
        'aws_access_key_id': access_key,
        'aws_secret_access_key': secret_key,
    }
    connection = boto3.client(service_name='sts', **credential_kwargs)
```

Reply:
you are correct, bit more easy to read and simpler